### ISSUE TITLE
Throws error if comment contains '--'

### ISSUE THREAD

**Issue by ejfrancis** [intuit/node-pom-parser#13]
If given an invalid pom file with a comment like this

```
<!-- some comment -- with dashes inside -->
```

the `parse()` methods throws an error. Since `parse()` returns a Promise, I would expect this throw to be caught internally and passed to `reject()` so that I can access it like the example below, but the error is actually thrown so I can't catch it with `.catch()` of the Promise

```javascript
parse()
.catch((e) => {
  // e is the thrown error
});
```

What's worse is that it seems to throw internally after some async action, so that you can't catch the error on the current tick of the event loop. So it's actually impossible to catch it, wrapping it in try/catch like below doesn't  catch the error

```javascript
try {
  parse()
catch (e) {
  // doesnt get caught here
}
```

**Comment by ejfrancis** [comment]
This is still broken a year later

**Comment by foobar98** [comment]
@nsinghal12 I don't think this issue is valid (anymore). Looking at the code, `parse` does not return a promise. It returns the error/response as part of the callback passed. I believe this can be closed. Please check.

**Comment by rylim** [comment]
Hi @nsinghal12 if no one is working on this issue may I get a try at it?

**Comment by nsinghal12** [comment]
Closing this issue as this will be addressed as part of #22
